### PR TITLE
fix: move blocks update all api

### PR DESCRIPTION
**Pull request by storyworks**
# Description

updateAll baseService doesn't properly update with objects with ids, only objects with keys. Currently the array we send from block.resolver have ids.

Refactored move logic to block.service. As Mike said "the KeyAsId/IdAsKey transformations are now handled in the services, instead of the resolvers. Keeping id in resolver and key in service."

-[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787371/todos/4786305141)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Manually testing move works
- [x] Deleting a block with children still works

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-journeys/src/app/modules/block/block.resolver.spec.ts [hunk 1]
```diff
@@ -89,53 +88,18 @@ describe('BlockResolver', () => {
...
     it('updates the block order', async () => {
-      await resolver.blockOrderUpdate('image1', '2', 2)
-      expect(service.reorderSiblings).toHaveBeenCalledWith([
-        image2,
-        image3,
-        image1
-      ])
-    })
-
-    it('does not update if block not part of current journey', async () => {
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/block/block.resolver.spec.ts:100]: Moved this test and below test to `block.service.spec.ts`

Comment (storyworks) [inline apps/api-journeys/src/app/modules/block/block.resolver.spec.ts:100]: Moved this test and below test to `block.service.spec.ts`

File: apps/api-journeys/src/app/modules/block/block.resolver.ts [hunk 2]
```diff
@@ -76,18 +54,4 @@ export class BlockResolver {
       parentBlockId
     )
   }
-
-  async siblings(@Parent() block: Block): Promise<Block[]> {
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/block/block.resolver.ts:80]: These 2 not used anywhere, used to be used in old `blockOrderUpdate` logic

File: apps/api-journeys/src/app/modules/block/block.service.spec.ts [hunk 3]
```diff
@@ -170,26 +170,47 @@ describe('BlockService', () => {
     })
   })
 
-  describe('updateChildrenParentOrder', () => {
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/block/block.service.spec.ts:173]: Removed this function below

File: apps/api-journeys/src/app/modules/block/block.service.ts [hunk 4]
```diff
@@ -40,28 +40,20 @@ export class BlockService extends BaseService {
...
     return await res.all()
   }
 
-  async updateChildrenParentOrder(
-    journeyId: string,
-    parentBlockId: string
-  ): Promise<Block[]> {
-    const siblings = await (
-      await this.getSiblingsInternal(journeyId, parentBlockId)
-    ).filter((block) => block.parentOrder != null)
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/block/block.service.ts:61]: Removed this function as filter is unneeded now, and remaining lines can be handled within the `removeBlockAndChildren` function (only place used).

Also `reorderBlock` calls `getSiblingsInternal` and `reorderSiblings` so keeps the pattern.

File: apps/api-journeys/src/app/modules/block/block.service.ts [hunk 5]
```diff
@@ -72,6 +64,27 @@ export class BlockService extends BaseService {
     return await this.updateAll(updatedSiblings)
   }
 
+  @KeyAsId()
+  async reorderBlock(
+    id: string,
+    journeyId: string,
+    parentOrder: number
+  ): Promise<Block[]> {
+    const block = idAsKey(await this.get(id)) as Block
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/block/block.service.ts:73]: `.get()` returns with id, convert to key so we send an array with all keys to `.updateAll()`

**Review by tataihono — APPROVED** [review]

**Pull request merged**